### PR TITLE
better tablename validation

### PR DESCRIPTION
**Pull request by clockfort**
note that I didn't force everyone to also fit inside the minimum of all of the allowed KVS table name identifier lengths; older Oracle identifier lengths are too abusively small and there are too many people who already have long table names.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3159)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java [hunk 1]
```diff
@@ -97,10 +102,11 @@ public void createTables(Map<TableReference, byte[]> tableRefToTableMetadata) {
     protected void sanityCheckTableName(TableReference tableRef) {
         String tableName = tableRef.getQualifiedName();
         Validate.isTrue(
-                (!tableName.startsWith("_") && tableName.contains("."))
+                (!tableName.startsWith("_") && (tableName.contains(".")))
+                        && ALLOWED_TABLE_NAMES.matcher(tableName).matches()
                         || AtlasDbConstants.hiddenTables.contains(tableRef)
                         || tableName.startsWith(AtlasDbConstants.NAMESPACE_PREFIX),
-                "invalid tableName: %s", tableName);
```
Comment (clockfort) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java]: this was just wrong before, Validate doesn't take format strings.
Not sure if this should also be fixed for safe logging, though

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java [hunk 2]
```diff
@@ -97,10 +102,11 @@ public void createTables(Map<TableReference, byte[]> tableRefToTableMetadata) {
     protected void sanityCheckTableName(TableReference tableRef) {
         String tableName = tableRef.getQualifiedName();
         Validate.isTrue(
-                (!tableName.startsWith("_") && tableName.contains("."))
+                (!tableName.startsWith("_") && (tableName.contains(".")))
+                        && ALLOWED_TABLE_NAMES.matcher(tableName).matches()
```
Comment (sandorw) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java]: nit: would prefer parentheses to make the operator precedence clearer (as with the existing && check)

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -87,6 +87,11 @@ v0.83.0
            Some exceptions thrown due to schema mutation failures now have type ``UncheckedExecutionException``.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/3152>`__)
 
+    *    - |changed|
```
Comment (sandorw) [inline docs/source/release_notes/release-notes.rst]: Put this up with the develop changes

**Comment by sandorw** [comment]
@clockfort was this sparked by a particular issue you hit? I'm curious if we have any backcompat concerns to worry about here that you know of - existing table names that don't match these rules.

**Comment by clockfort** [comment]
dropping because no one cares about this issue and I want sandor to review my other PR 🐌

**Pull request closed without merging**